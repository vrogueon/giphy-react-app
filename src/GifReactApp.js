import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifReactApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState([defaultCategories]);

    // const handleAdd = () => {
    //     setCategories([...categories, 'YuYu Hakusho']);
    // }

    return (
        <>
            <h2>GifReactApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            <ol>
                {
                    categories.map(category => 
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    )
                }
            </ol>
        </>
    )
}
