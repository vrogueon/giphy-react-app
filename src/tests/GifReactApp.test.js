import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme/build';
import { GifReactApp } from '../GifReactApp';

describe('pruebas en GifReactApp', () => {
    
    test('debe de hacer match con el snapshot', () => {
        
        const wrapper = shallow(<GifReactApp/>);
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['jjba'];
        const wrapper = shallow(<GifReactApp defaultCategories={ categories }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
});''
