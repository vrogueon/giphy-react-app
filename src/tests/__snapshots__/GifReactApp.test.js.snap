// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifReactApp debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifReactApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={Array []}
      key=""
    />
  </ol>
</Fragment>
`;

exports[`pruebas en GifReactApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifReactApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category={
        Array [
          "jjba",
        ]
      }
      key="jjba"
    />
  </ol>
</Fragment>
`;
